src/playlist-loader/dash-media-playlist-loader.js

```diff
@@ -0,0 +1,235 @@
+import PlaylistLoader from './playlist-loader.js';
+import containerRequest from '../util/container-request.js';
+import {addSidxSegmentsToPlaylist} from 'mpd-parser';
+import parseSidx from 'mux.js/lib/tools/parse-sidx';
+import {toUint8} from '@videojs/vhs-utils/es/byte-helpers';
+import {segmentXhrHeaders} from '../xhr';
+import {mergeMedia, forEachPlaylist} from './utils.js';
+
+/**
+ * This function is used internally to keep DashMainPlaylistLoader
+ * up to date with changes from this playlist loader.
+ *
+ * @param {Object} mainManifest
+ *        The manifest from DashMainPlaylistLoader
+ *
+ * @param {string} uri
+ *        The uri of the playlist to find
+ *
+ * @return {null|Object}
+ *         An object with get/set functions or null.
+ */
+export const getMediaAccessor = function(mainManifest, uri) {
+  let result = null;
+
+  forEachPlaylist(mainManifest, function(media, index, array) {
+    if (media.uri === uri) {
+      result = {
+        get: () => array[index],
+        set: (v) => {
+          array[index] = v;
+        }
+      };
+
+      return true;
+    }
+  });
+
+  return result;
+};
+
+/**
+ * A class to encapsulate all of the functionality for
+ * Dash media playlists. Note that this PlaylistLoader does
+ * not refresh, parse, or have manifest strings. This is because
+ * Dash doesn't really have media playlists. We only use them because:
+ * 1. We want to match our HLS API
+ * 2. Dash does have sub playlists but everything is updated on main.
+ *
+ * @extends PlaylistLoader
+ */
+class DashMediaPlaylistLoader extends PlaylistLoader {
```

Comment:
This class deals:
1. finding it's playlist whenever `DashMainPlaylistLoader` triggers `updated`
2. Requesting sidx and adding sidx segments
3. Setting the media refresh time on `DashMainPlaylistLoader` so that it can refresh using target duration.